packages/common/http/src/fetch.ts

```
@@ -0,0 +1,176 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {Injectable, Optional} from '@angular/core';
+import {EMPTY, forkJoin, from, Observable, Observer, of, throwError} from 'rxjs';
+import {catchError, finalize, map, mergeMap, startWith} from 'rxjs/operators';
+
+import {HttpBackend} from './backend';
+import {HttpHeaders} from './headers';
+import {HttpRequest} from './request';
+import {HttpErrorResponse, HttpEvent, HttpEventType, HttpResponse, HttpSentEvent, HttpStatusCode} from './response';
+
+const XSSI_PREFIX = /^\)\]\}',?\n/;
+
+/**
+ * Determine an appropriate URL for the response, by checking either
+ * response url or the X-Request-URL header.
+ */
+function getResponseUrl(response: Response): string|undefined {
+  if (response.url) {
+    return response.url;
+  }
+
+  // stored as lowercase in the map
+  const xRequestUrl = 'X-Request-URL'.toLocaleLowerCase();
+  return response.headers.get(xRequestUrl) ?? undefined;
+}
+
+/**
+ * Uses `fetch` to send requests to a backend server.
+ * This backend requires nodejs >= 18
+ * @see {@link HttpHandler}
+ *
+ * @developerPreview
+ */
+@Injectable()
+export class FetchBackend implements HttpBackend {
+  private fetchImpl: typeof fetch;
+
+  // Allow to pass a custom implementation for `fetch()`, a mocked fetch for example
+  public constructor(@Optional() fetchFactory?: FetchFactory) {
+    this.fetchImpl = fetchFactory?.fetch ?? fetch;
+  }
+
+  handle(req: HttpRequest<any>): Observable<HttpEvent<any>> {
+    const abortController = new AbortController();
+    const init = this.createRequestInit(req);
+    return from(this.fetchImpl(req.url, {signal: abortController.signal, ...init}))
+        .pipe(
+            catchError((error) => {
```

In case the request is aborted, we should throw an error, thus we need to filter here.

`!(error instanceof DOMException && error.name === 'AbortError')`

---

I'm not sure I understand what you mean.  
To me the implementations matches what we have with the `HttpXhrBackend` and is covered by this test : 

```typescript
    it('emits an error when browser cancels a request', done => {
      backend.handle(TEST_POST).subscribe({
        error: (err: HttpErrorResponse) => {
          expect(err instanceof HttpErrorResponse).toBe(true);
          expect(err.error instanceof DOMException).toBeTruthy();
          done();
        }
      });
      fetchMock.mockAbortEvent();
    });
```

Or did I miss something  ?

---

Oh.. interesting. I missed that.